### PR TITLE
Inbox api

### PR DESCRIPTION
**Pull request by ftcvlad**
Add APIs and models to support building out an in-app message inbox UI.

### CODE REVIEW

File: kumulos/src/main/java/com/kumulos/android/InAppMessageService.java [hunk 1]
```diff
@@ -189,4 +189,42 @@ private static void storeLastSyncTime(Context context, List<InAppMessage> inAppM
...
+
+    static List<InAppInboxItem> readInboxItems(Context context){
+        Callable<List<InAppInboxItem>> task = new InAppContract.ReadInAppInboxCallable(context);
+        final Future<List<InAppInboxItem>> future = Kumulos.executorService.submit(task);
+
+        List<InAppInboxItem> inboxItems;
+        try {
+            inboxItems = future.get();
+        } catch (InterruptedException | ExecutionException ex) {
+            return null;
```
Comment (cgwyllie) [inline kumulos/src/main/java/com/kumulos/android/InAppMessageService.java]: Easier for consumer if we return an empty list here?

File: kumulos/src/main/java/com/kumulos/android/InAppMessageService.java [hunk 2]
```diff
@@ -189,4 +189,42 @@ private static void storeLastSyncTime(Context context, List<InAppMessage> inAppM
...
+
+        InAppMessage inboxMessage;
+        try {
+            inboxMessage = future.get();
+        } catch (InterruptedException | ExecutionException ex) {
+            return KumulosInApp.InboxMessagePresentationResult.FAILED;
+        }
+
+        if (inboxMessage == null){
+           return KumulosInApp.InboxMessagePresentationResult.FAILED_EXPIRED;
```
Comment (cgwyllie) [inline kumulos/src/main/java/com/kumulos/android/InAppMessageService.java]: Is null only possible due to expiry, wonder if better to test in code when handed the item to present?

File: kumulos/src/main/java/com/kumulos/android/Kumulos.java [hunk 3]
```diff
@@ -63,9 +62,7 @@
     /** package */ static String authHeader;
     /** package */ static ExecutorService executorService;
     private static final Object userIdLocker = new Object();
-    private static Application application;
-    private static InAppActivityLifecycleWatcher inAppActivityWatcher;
-    static InAppDeepLinkHandlerInterface inAppDeepLinkHandler = null;
+    static Application application;
```
Comment (cgwyllie) [inline kumulos/src/main/java/com/kumulos/android/Kumulos.java]: Seems the Kumulos class doesn't use this ref anymore, can/should it move with the in-app methods to KumulosInApp?

Comment (ftcvlad) [reply kumulos/src/main/java/com/kumulos/android/Kumulos.java]: true

File: kumulos/src/main/java/com/kumulos/android/InAppMessageService.java (hunk 2 above)
Comment (ftcvlad) [reply kumulos/src/main/java/com/kumulos/android/InAppMessageService.java]: now return EXPIRED only when sure it is expired. When message is null, return FAILED

**Pull request merged**